test/evp_test.c

```diff
@@ -865,18 +865,26 @@ static int cipher_test_enc(EVP_TEST *t, int enc, size_t out_misalign,
 
     /* Test that the cipher dup functions correctly if it is supported */
     ERR_set_mark();
-    if (EVP_CIPHER_CTX_copy(ctx, ctx_base)) {
-        EVP_CIPHER_CTX_free(ctx_base);
-        ctx_base = NULL;
-    } else {
-        EVP_CIPHER_CTX_free(ctx);
-        ctx = ctx_base;
+    if (!EVP_CIPHER_CTX_copy(ctx, ctx_base)) {
+        if (fips_provider_version_ge(libctx, 3, 0, 0)) {
+            TEST_info("Doing a copy of Cipher %s Fails!\n", EVP_CIPHER_get0_name(expected->cipher));
+            ERR_print_errors_fp(stderr);
+            goto err;
+        } else
+            TEST_info("Allowing failed copy as fips provider not present or not in use\n");
```

Comment:
The message is misleading. Correct would be `Allowing copy failure as old fips provider is in use.\n`